internal/services/serviceprincipals/service_principal_claims_mapping_policy_assignment.go

```diff
@@ -0,0 +1,142 @@
+package serviceprincipals
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azuread/internal/clients"
+	"github.com/hashicorp/terraform-provider-azuread/internal/services/serviceprincipals/parse"
+	"github.com/hashicorp/terraform-provider-azuread/internal/tf"
+	"github.com/manicminer/hamilton/msgraph"
+)
+
+func servicePrincipalClaimsMappingPolicyAssignment() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: servicePrincipalClaimsMappingPolicyAssignmentResourceCreate,
+		ReadContext:   servicePrincipalClaimsMappingPolicyAssignmentResourceRead,
+		DeleteContext: servicePrincipalClaimsMappingPolicyAssignmentResourceDelete,
+
+		Importer: tf.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.ObjectSubResourceID(id, "azuread_claims_mapping_policy")
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"claims_mapping_policy_id": {
+				Description: "ID of the claims mapping policy to assign",
+				ForceNew:    true,
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+
+			"service_principal_id": {
+				Description: "ID of the service principal to assign the policy to",
+				ForceNew:    true,
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+		},
+	}
+}
+
+func servicePrincipalClaimsMappingPolicyAssignmentResourceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client).ServicePrincipals.ServicePrincipalsClient
+
+	claimID := d.Get("claims_mapping_policy_id").(string)
+
+	assignment := []msgraph.ClaimsMappingPolicy{
+		{
+			ID: &claimID,
+		},
+	}
+
+	spID := d.Get("service_principal_id").(string)
+
+	sp := msgraph.ServicePrincipal{
+		DirectoryObject: msgraph.DirectoryObject{
+			ID: &spID,
+		},
+		ClaimsMappingPolicies: &assignment,
+	}
+	_, err := client.AssignClaimsMappingPolicy(ctx, &sp)
```

Comment:
I think this would be better condensed into

```suggestion
	properties := msgraph.ServicePrincipal{
		DirectoryObject: msgraph.DirectoryObject{
			ID: utils.String(d.Get("service_principal_id").(string)),
		},
		ClaimsMappingPolicies: &[]msgraph.ClaimsMappingPolicy{
			{
				ID: utils.String(d.Get("claims_mapping_policy_id").(string)),
			},
		},
	}
	_, err := client.AssignClaimsMappingPolicy(ctx, &properties)
```

Reply:
Adjusted